sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/UploadOutputStreamTest.java

```diff
@@ -3,29 +3,25 @@
 
 package com.azure.storage.blob.perf;
 
-import static com.azure.perf.test.core.TestDataCreationHelper.createRandomInputStream;
-
 import com.azure.perf.test.core.PerfStressOptions;
 import com.azure.storage.blob.perf.core.BlobTestBase;
 import com.azure.storage.blob.specialized.BlobOutputStream;
-import java.io.IOException;
-import java.io.InputStream;
 import reactor.core.publisher.Mono;
 
-public class UploadOutputStreamTest extends BlobTestBase<PerfStressOptions> {
+import java.io.IOException;
 
-    private final InputStream inputStream;
+import static com.azure.perf.test.core.TestDataCreationHelper.writeBytesToOutputStream;
 
+public class UploadOutputStreamTest extends BlobTestBase<PerfStressOptions> {
     public UploadOutputStreamTest(PerfStressOptions options) {
         super(options);
-        this.inputStream = createRandomInputStream(options.getSize());
     }
 
     @Override
     public void run() {
         try {
             BlobOutputStream blobOutputStream = blockBlobClient.getBlobOutputStream();
-            copyStream(inputStream, blobOutputStream);
+            writeBytesToOutputStream(blobOutputStream, options.getSize());
```

Comment:
Would this be equivalent in both perf and functionality?

```java
copyStream(createRandomInputStream(size), blobOutputStream);
```

If so, this would allow us to delete `writeBytesToOutputStream()` which seems simpler.

Reply:
`copyStream` is generally less performant, and definitely less performant with the type of streams we are using. `ByteArrayInputStream` clones the `byte[]` it returns from `read` operations and `ByteArrayOutputSteam` clones the `byte[]` passed into `write` operation. `writeBytesToOutputStream` skips the data copying that occurs during `read`.